### PR TITLE
Atomic transaction for the `migrate_transcripts` command

### PR DESCRIPTION
**Pull request by OmarIthawi**
A course will either be successful or will fail in one atomic SQL transaction.

Jira: RED-2483.

### TODO
 - [x] Add tests.
 - [x] Test on staging, no exceptions were found.

### CODE REVIEW

File: cms/djangoapps/contentstore/tasks.py [hunk 1]
```diff
@@ -108,8 +108,17 @@ def enqueue_async_migrate_transcripts_tasks(course_keys,
         'command_run': command_run
     }
 
+    from django.db import transaction
```
Comment (thraxil) [inline cms/djangoapps/contentstore/tasks.py:111]: should we keep imports together at the top of the file?

**Review by thraxil — APPROVED** [review]

**Review by estherjsuh — APPROVED** [review]

File: cms/djangoapps/contentstore/tasks.py [hunk 2]
```diff
@@ -108,8 +108,18 @@ def enqueue_async_migrate_transcripts_tasks(course_keys,
...
 
+    # TODO: Undo all the migrate_transcripts edits: https://github.com/appsembler/edx-platform/issues/1068
+    from django.db import transaction  # temp import, kept here to reduce migration conflicts
+
     for course_key in course_keys:
-        async_migrate_transcript(text_type(course_key), **kwargs)
+        try:
+            with transaction.atomic():
+                async_migrate_transcript(text_type(course_key), **kwargs)
+        except Exception:
```
Comment (OmarIthawi) [inline cms/djangoapps/contentstore/tasks.py:118]:
```suggestion
        except ItemNotFoundError:
            LOGGER.info(
                '[%s] [run=%s] [course-transcript-migration-failed-with-not-found-error-exc] [course=%s]',
                MIGRATION_LOGS_PREFIX, command_run, course_key
            )
        except Exception:
```

File: cms/djangoapps/contentstore/tasks.py (hunk 1 above)
Comment (OmarIthawi) [reply cms/djangoapps/contentstore/tasks.py:113]: In general yes. I this case, I'm planning to revert this hack entirely, so it's worth to keep everything in a single place.

**Comment by OmarIthawi** [comment]
@shadinaif @melvinsoft @estherjsuh I'm going to merge this pull request _without tests_. Please check https://github.com/appsembler/edx-platform/pull/1067#issue-1113708160 for more details.

I'll wait till Wednesday and merge unless someone objects.

**Pull request merged**